File: Backend/src/main/java/com/bachld/project/backend/dto/response/decuong/DeCuongLogResponse.java
package com.bachld.project.backend.dto.response.decuong;

import com.bachld.project.backend.enums.DeCuongState;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.util.List;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
public class DeCuongLogResponse {
    String fileUrlMoiNhat;
    LocalDate ngayNopGanNhat;
    Integer tongSoLanNop;
    DeCuongState trangThaiHienTai;
    List<RejectNote> cacNhanXetTuChoi; // chỉ các lần bị từ chối trong quá khứ

    @Getter @Setter
    @NoArgsConstructor @AllArgsConstructor
    @FieldDefaults(level = lombok.AccessLevel.PRIVATE)
    public static class RejectNote {
        LocalDate ngayNhanXet; // createdAt của log
        String lyDo;
    }
}

File: Android/app/src/main/java/com/bachld/android/data/dto/response/decuong/DeCuongLogResponse.kt
package com.bachld.android.data.dto.response.decuong

data class DeCuongLogResponse(
    val fileUrlMoiNhat: String?,
    val ngayNopGanNhat: String?,       // yyyy-MM-dd từ LocalDate
    val tongSoLanNop: Int?,
    val trangThaiHienTai: DeCuongState?,
    val cacNhanXetTuChoi: List<RejectNote>?
) {
    data class RejectNote(
        val ngayNhanXet: String?,      // yyyy-MM-dd
        val lyDo: String?
    )
}
